### ISSUE TITLE
small pairing optimization/`precompute_miller_lines`

### ISSUE THREAD

**Issue by laudiacay** [laudiacay/barustenberg#57]
zac added `precompute_miller_lines` as a VERY small optimization to the pairing code.

We have about 99% of the code needed to make this work implemented in the `srs` branch! However, last second we chose to use arkworks for this, because the last 1% of the work would have been annoying.

There's a missing constant in arkworks that's necessary here to implement our own pairing from scratch. It's `twist_coeff_b` from the Fq2 params in the original code. What is it? We (including zac) think it's the (Fq2) `b` of the Weierstrass form of the `Fq12` twist's automorphism curve, represented as a pair of `x` and `y` `Fq`s.

It's constructed in a weird representation (some kind of limb form?) in Barettenberg. We don't know what this representation is yet. So far we've been unable to figure out how to get to the arkworks representation of the constants (just a decimal string representing x of the point) to turn into the barettenberg constants that we do have. It doesn't seem to be an endianness issue. It might be a WNAF or montgomery representation situation. We don't know. Zac recommended we get the c++ compiling and print things out... 

tl;dr holding off on this! it should be a VERY small performance gain. We'll get back to it FAR in the future.

CC after discussion with @Autoparallel
